babycare_partner_street_number/models/res_partner.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.one
+    @api.depends('street_name', 'street_number', 'street_number_addition')
+    def _get_street(self):
+        self.street = ' '.join(
+            filter(None, [self.street_name, self.street_number,
+                          self.street_number_addition]))
+
+    def _write_street(self):
+        return False
```

Comment:
Any write on the 'street' field will now be dropped silently. Are you sure that this is safe? Possible data loss. You might want to raise an exception instead (although that might lead to blocking situations in your workflow). Having the original functionality from partner_street_number as a fallback still seems like a pretty good idea to me.

For existing addresses, you are too late removing the functionality here. Fortunately, the sanity limit applied on detected house numbers with a maximum length of 5 in the original method fits neatly to your length limit on that field.

Reply:
I tested the original functionality from partner_street_number and this functionality is indeed great for the first import for example. But unfortunately it messes with the street, number and addition also afterwards. Let me give you an example.

- I created a Test Customer with the address Street (in street_name field) 32 (in street_number field) Bis (in street_number_addition field). ![babycare_partner_street_number_1](https://user-images.githubusercontent.com/10392890/30798549-579b0228-a1da-11e7-9a19-f223a5592459.JPG)

- After I hit save the _write_street method is 'styling' the address like this: 
![babycare_partner_street_number_2](https://user-images.githubusercontent.com/10392890/30798605-8b5a1554-a1da-11e7-86fa-594fe3b8591e.JPG)

- As you can see in the edit-mode the 32 and Bis is added to the street in street_name but the 'Bis' is also still available in his own street_number_addition field. The street_number field is empty. 
![babycare_partner_street_number_3](https://user-images.githubusercontent.com/10392890/30798675-c1c314a6-a1da-11e7-9db0-ce65674884d4.JPG)

- This is the behavior after every change in one of the fields of street_name, street_number or street_number_addition. So for example if add a new number (36) in the street_number field but keep the existing number (32) in the street_name field the outcome is as follows:
![babycare_partner_street_number_4](https://user-images.githubusercontent.com/10392890/30798742-fc03338a-a1da-11e7-8151-8480869fdbd6.JPG)

As you can see the _write_street method in the original module partner_street_number 'messes' with the outcome of the street fields. Therefore, I'd like to disable the _write_street method after the first install of the original module partner_street_number. I think my 'return False' proposal is a solution for this? But if you have a better solution I'd like to hear that!

Reply:
Another important note: for the PostNL courier export it is necessary that all fields (street, street_number and street_number_addition (if applicable)) are filled. The rewrite is not usable in combination with the PostNL courier export.

Reply:
I see. It is a bug. In a form that properly displays the street name and street number fields, this method should not be called at all. It is purely meant as backwards compatibility for other addons that write to the street field directly. Attempted fix here: https://github.com/OCA/partner-contact/pull/478/files

Reply:
It's now https://github.com/OCA/partner-contact/pull/479